test/cmd/legacy-script.sh

```diff
@@ -15,6 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# shellcheck disable=SC2034
+# https://github.com/koalaman/shellcheck/wiki/SC2034
```

Comment:
why are we disabling this here but not in the 1400 line commit?

Reply:
That big commit is about exported vars, this fix is shellchecks related and specific to current file.
And at first, I put it in the wrong place and breaks the boilerplate header. Move it to here in a fix commit.